lang/csharp/src/apache/main/CodeGen/CodeGenUtil.cs

```diff
@@ -70,6 +71,9 @@ public sealed class CodeGenUtil
         private const char At = '@';
         private const char Dot = '.';
 
+        private readonly string _assemblyInformationVersion = FileVersionInfo.GetVersionInfo(
```

Comment:
Keep in mind that FileVersionInfo.GetVersionInfo will open the actual dll file and read the library file directly to figure out the version + you will need the using Diagnostics.
https://source.dot.net/#System.Diagnostics.FileVersionInfo/System/Diagnostics/FileVersionInfo.Unix.cs,45

System.Reflection.Assembly.GetExecutingAssembly().GetCustomAttribute<AssemblyInformationalVersionAttribute>().InformationalVersion is more efficient.